clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2439,7 +2438,10 @@ void handleResponse(AbstractResponse abstractResponse) {
 
                         @Override
                         void handleFailure(Throwable throwable) {
-                            completeAllExceptionally(futures.values(), throwable);
+                            for (TopicPartition tp : entry.getValue().keySet()) {
```

Comment:
If you don't mind, let's defer this to https://github.com/apache/kafka/pull/7449, just so we have the JIRA to go along with it.

Reply:
Sure, I'll undo this bit. I'll push an update in the next 15mins with some of the tests